sdk/core/System.ServiceModel.Rest/src/Temporary/RequestUri.cs

```diff
@@ -0,0 +1,531 @@
+#endif
+            }
+            else
+            {
+#if NETCOREAPP2_1_OR_GREATER
+                _pathAndQuery.Append(path);
+#else
+                _pathAndQuery.Append(stringPath);
+#endif
+            }
+        }
+
+        /// <summary>
+        /// Returns a string representation of this <see cref="RequestUri"/>.
+        /// </summary>
+        /// <returns>A string representation of this <see cref="RequestUri"/>.</returns>
+        public override string ToString()
+        {
+            var stringBuilder = new StringBuilder(
+                (Scheme?.Length ?? 0) +
+                3 + // ://
+                (Host?.Length ?? 0) +
+                _pathAndQuery.Length +
+                10 // optimistic padding
+                );
+            stringBuilder.Append(Scheme);
+            stringBuilder.Append("://");
+            stringBuilder.Append(Host);
+            if (!HasDefaultPortForScheme)
+            {
+                stringBuilder.Append(':');
+                stringBuilder.Append(Port);
+            }
+
+            if (_pathAndQuery.Length == 0 || _pathAndQuery[0] != PathSeparator)
+            {
+                stringBuilder.Append(PathSeparator);
+            }
+
+            stringBuilder.Append(_pathAndQuery);
+
+            return stringBuilder.ToString();
+        }
+
+        private bool HasDefaultPortForScheme =>
+            (Port == 80 && string.Equals(Scheme, "http", StringComparison.InvariantCultureIgnoreCase)) ||
+            (Port == 443 && string.Equals(Scheme, "https", StringComparison.InvariantCultureIgnoreCase));
+
+        private void ResetUri()
+        {
+            _uri = null;
+        }
+        #endregion
+
+        #region RawRequestUriBuilder
+
+        private const string SchemeSeparator = "://";
+        private const char HostSeparator = '/';
+        private const char PortSeparator = ':';
+        private static readonly char[] HostOrPort = { HostSeparator, PortSeparator };
+        private const char QueryBeginSeparator = '?';
+        private const char QueryContinueSeparator = '&';
+        private const char QueryValueSeparator = '=';
+
+        private RawWritingPosition? _position;
+
+        private static void GetQueryParts(ReadOnlySpan<char> queryUnparsed, out ReadOnlySpan<char> name, out ReadOnlySpan<char> value)
+        {
+            int separatorIndex = queryUnparsed.IndexOf(QueryValueSeparator);
+            if (separatorIndex == -1)
+            {
+                name = queryUnparsed;
+                value = ReadOnlySpan<char>.Empty;
+            }
+            else
+            {
+                name = queryUnparsed.Slice(0, separatorIndex);
+                value = queryUnparsed.Slice(separatorIndex + 1);
+            }
+        }
+
+        /// <summary>
+        /// TBD.
+        /// </summary>
+        /// <param name="value"></param>
+        /// <param name="escape"></param>
+        public void AppendRawPathOrQueryOrHostOrScheme(string value, bool escape)
```

Comment:
This name makes us all cry, it is true.  It is this way to prevent the RawRequestUriBuilder shared source file from having an inherited member name clash across versions.  When we're able to remove this file, we can clean this up.